worlds/aquaria/__init__.py

```diff
@@ -0,0 +1,222 @@
+            # The last two are inverted because in the original game, they are special turtle that communicate directly
+            self.__pre_fill_item("Transturtle Simon says", "Arnassi Ruins, Transturtle", precollected)
+            self.__pre_fill_item("Transturtle Arnassi ruins", "Simon says area, Transturtle", precollected)
+        for name, data in item_table.items():
+            if name in precollected:
+                precollected.remove(name)
+                self.multiworld.itempool.append(self.create_item(self.get_filler_item_name()))
+            else:
+                if name not in self.exclude:
+                    for i in range(data[1]):
+                        item = self.create_item(name)
+                        self.multiworld.itempool.append(item)
+
+    def set_rules(self) -> None:
+        """
+        Launched when the Multiworld generator is ready to generate rules
+        """
+
+        self.regions.adjusting_rules(self.options)
+        self.multiworld.completion_condition[self.player] = lambda \
+            state: state.has("Victory", self.player)
+
+    def generate_basic(self) -> None:
+        """
+        Player-specific randomization that does not affect logic.
+        Used to fill then `ingredients_substitution` list
+        """
+        simple_ingredients_substitution = [i for i in range(27)]
+        if self.options.ingredient_randomizer.value > 0:
+            if self.options.ingredient_randomizer.value == 1:
+                simple_ingredients_substitution.pop(-1)
+                simple_ingredients_substitution.pop(-1)
+                simple_ingredients_substitution.pop(-1)
+            self.random.shuffle(simple_ingredients_substitution)
+            if self.options.ingredient_randomizer.value == 1:
+                simple_ingredients_substitution.extend([24, 25, 26])
+        dishes_substitution = [i for i in range(27, 76)]
+        if self.options.dish_randomizer:
+            self.random.shuffle(dishes_substitution)
+        self.ingredients_substitution.clear()
+        self.ingredients_substitution.extend(simple_ingredients_substitution)
+        self.ingredients_substitution.extend(dishes_substitution)
+
+    def fill_slot_data(self) -> Dict[str, Any]:
+        aquarian_translation = False
+        if self.options.aquarian_translation:
+            aquarian_translation = True
+        skip_first_vision = False
+        if self.options.skip_first_vision:
+            skip_first_vision = True
+        return {"ingredientReplacement": self.ingredients_substitution,
+                "aquarianTranslate": aquarian_translation,
+                "secret_needed": self.options.objective.value > 0,
+                "minibosses_to_kill": self.options.mini_bosses_to_beat.value,
+                "bigbosses_to_kill": self.options.big_bosses_to_beat.value,
+                "skip_first_vision": skip_first_vision,
+                }
```

Comment:
Because options are not really boolean. They are object that translate to true and false with some internal mecanics. But if I return the value directly, the json serializer tell that it cannot serialize an option. So, I have to translate the option to boolean so that the serialiser know what to do.

Reply:
Just for the record, here is the error that I get
```Object of type AquarianTranslation is not JSON serializable
Traceback (most recent call last):
  File "/home/louis/prog/aquaria/Archipelago/MultiServer.py", line 750, in server
    await process_client_cmd(ctx, client, msg)
  File "/home/louis/prog/aquaria/Archipelago/MultiServer.py", line 1685, in process_client_cmd
    await ctx.send_msgs(client, reply)
  File "/home/louis/prog/aquaria/Archipelago/MultiServer.py", line 286, in send_msgs
    msg = self.dumper(msgs)
  File "/home/louis/prog/aquaria/Archipelago/NetUtils.py", line 105, in encode
    return _encode(_scan_for_TypedTuples(obj))
  File "/usr/lib/python3.10/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python3.10/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/usr/lib/python3.10/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type AquarianTranslation is not JSON serializable
``

Reply:
But it seems to work with the bool(...). Good thinking.